insights/parsers/package_provides_java.py

```diff
@@ -0,0 +1,36 @@
+"""
+PackageProvidesJava - Command ``rpm -qf``
+=========================================
+
+This parser handles ``rpm -qf`` output.
+
+Typical contents of the ``rpm -qf package`` command::
+
+    java-1.8.0-openjdk-1.8.0.144-0.b01.el7_4.x86_64
+
+Examples:
+
+    >>> pack_pro_java = shared[PackageProvidesJava]
+    >>> pack_pro_java.java_package
+    "java-1.8.0-openjdk-headless-1.8.0.144-0.b01.el7_4.x86_64"
+
+"""
+
+from insights import parser, Parser
+
+
+@parser('package_provides_java')
+class PackageProvidesJava(Parser):
+    def parse_content(self, content):
+        """
+        Parse the output of the ``rpm -qf package`` command.
+        """
+        self.data = None
+        for line in content:
+            if 'not owned by any package' not in line:
```

Comment:
If you are unable to determine the results here, then an exception should be raised instead of an empty string or `None`.  See the documentation on exceptions here: http://insights-core.readthedocs.io/en/latest/exception_model.html#parsing-and-exceptions